src/common/dcgraph.cpp

```diff
@@ -752,9 +752,16 @@ void wxGCDCImpl::DoDrawPoint( wxCoord x, wxCoord y )
     if (!m_logicalFunctionSupported)
         return;
 
-#if defined(__WXMSW__) && wxUSE_GRAPHICS_GDIPLUS
-    // single point path does not work with GDI+
-    if (m_graphicContext->GetRenderer() == wxGraphicsRenderer::GetGDIPlusRenderer())
+#if defined(__WXMSW__)
+    bool useStrokeLine = false;
+#if wxUSE_GRAPHICS_GDIPLUS
+    useStrokeLine |= m_graphicContext->GetRenderer() == wxGraphicsRenderer::GetGDIPlusRenderer();
+#endif
+#if wxUSE_GRAPHICS_DIRECT2D
+    useStrokeLine |= m_graphicContext->GetRenderer() == wxGraphicsRenderer::GetDirect2DRenderer();
+#endif
+    // single point path does not work with GDI+ and D2D
```

Comment:
I'm pretty sure single point path does work with D2D. What exactly is wrong with it?

Reply:
See below screenshots of the drawing sample were I added the following code to `DrawDefault` and changed the background colour to black:
```c++
    dc.SetPen(wxPenInfo(*wxGREEN, 10, wxPENSTYLE_SOLID));
    dc.DrawPoint(25, 25);
    dc.DrawPoint(50, 25);
```

You can see D2D draws nothing, and msw wxDC does not use the pen width.

I tested again and you're right that D2D does not need the GDI+ work-around (adding dx/dy), it can use `StrokeLine(x, y, x, y)` directly. The current implementation using `wxGraphicsPath` does not work.

msw wxDC - GDI+ - D2D
<a href="https://user-images.githubusercontent.com/8088070/62682741-2c9db000-b9bd-11e9-8bb1-72d18ee17fd5.png"><img src="https://user-images.githubusercontent.com/8088070/62682741-2c9db000-b9bd-11e9-8bb1-72d18ee17fd5.png" width="150" alt="msw wxDC"/></a><a href="https://user-images.githubusercontent.com/8088070/62682745-30313700-b9bd-11e9-9786-0611916a7c30.png"><img src="https://user-images.githubusercontent.com/8088070/62682745-30313700-b9bd-11e9-9786-0611916a7c30.png" width="150" alt="msw GDI+"/></a><a href="https://user-images.githubusercontent.com/8088070/62682748-31fafa80-b9bd-11e9-90f0-28a6b0ef08fe.png"><img src="https://user-images.githubusercontent.com/8088070/62682748-31fafa80-b9bd-11e9-90f0-28a6b0ef08fe.png" width="150" alt="msw D2D"/></a>

Ubuntu x11(?) wxDC - cairo - macOS core graphics
<a href="https://user-images.githubusercontent.com/8088070/62682762-36bfae80-b9bd-11e9-87d3-779bac44544b.png"><img src="https://user-images.githubusercontent.com/8088070/62682762-36bfae80-b9bd-11e9-87d3-779bac44544b.png" width="150" alt="gtk wxDC"/></a><a href="https://user-images.githubusercontent.com/8088070/62682765-38897200-b9bd-11e9-95bd-65c4197f5c8d.png"><img src="https://user-images.githubusercontent.com/8088070/62682765-38897200-b9bd-11e9-95bd-65c4197f5c8d.png" width="150" alt="gtk ciaro"/></a><a href="https://user-images.githubusercontent.com/8088070/62682770-3aebcc00-b9bd-11e9-81c2-457af01127c4.png"><img src="https://user-images.githubusercontent.com/8088070/62682770-3aebcc00-b9bd-11e9-81c2-457af01127c4.png" width="150" alt="macOS gc"/></a>

Reply:
I'm sure I tested all the `wxGraphicsContext` backends when I added `wxGCDCImpl::DoDrawPoint()`. I will investigate.

It looks like you tested with GTK3, where `wxDC` uses Cairo. If you try it with GTK2, I think you'll see that it behaves the same as MSW. Which is as described in the current documentation. My take on `DrawPoint()` is that it works properly with unit scale and pen width, anything else may or may not work.

Reply:
> It looks like you tested with GTK3

You're right. I updated the screenshot with GTK2 version. It's the same as wxDC on MSW.

What would you propose (besides fixed D2D)? Update `wxDC` to draw bigger points with penwidth >1. Or always draw a point of 1 pixel in `wxGraphicsContext`. Or just document the difference of DrawPoint between `wxDC` and `wxGCDC`?

Reply:
`DrawPoint()` did work with D2D when I checked it in, but was subsequently broken by c8fe811636

Reply:
This regression with D2D sub-path should be fixed in 283ac3096ddd431b19b36eeb903ba9ed31bcb674.

Reply:
Thanks. I'll revert my commit.